experiment/measurer/measure_worker.py

```diff
@@ -78,6 +80,49 @@ def extract_corpus(corpus_archive: str, sha_blacklist: Set[str],
         filesystem.write(file_path, member_contents, 'wb')
 
 
+class StateFile:
+    """A class representing the state of measuring a particular trial on
+    particular cycle. Objects of this class are backed by files stored in the
+    bucket."""
+
+    def __init__(self, name: str, state_dir: str, cycle: int):
+        self.name = name
+        self.state_dir = state_dir
+        self.cycle = cycle
+        self._prev_state = None
+
+    def _get_gcs_cycle_state_file_path(self, cycle: int) -> str:
```

Comment:
nit: does "gcs" needs to be in the function name, or could we start making these function name generic, so we won't have to rename them when we have local support?

Reply:
I changed name to bucket.